src/main/java/com/univ/lille/copainsderoute/platine/service/EventService.java

```diff
@@ -202,4 +205,34 @@ public List<EventResponseDTOs> getEventsByUserParticipated(String login) throws
     private List<EventResponseDTOs> removeFullEvents(List<EventResponseDTOs> events) {
         return events.stream().filter(Predicate.not(EventResponseDTOs::isFull)).toList();
     }
+
+    public List<EventResponseDTOs> getEventsByFilters(FilterEventRequestDto filterEventRequestDto, String login) throws UserNotFoundException, ZeroEventFoundException{
+        User user = userRepository.findByLogin(login).orElseThrow(UserNotFoundException::new);
+
+        List<Event> events = eventRepository.findByFilter(filterEventRequestDto.getStartDate(), filterEventRequestDto.getEndDate(), filterEventRequestDto.getDistanceMin(), filterEventRequestDto.getDistanceMax(), filterEventRequestDto.getVisibility(), filterEventRequestDto.getRoadTypes(), filterEventRequestDto.getBikeTypes());
+
+        List<Event> eventsFilteredVisibility = new ArrayList<>();
+        List<Friends> friends = user.getFriends();
+
+        for(Event e : events){
+            if(e.getVisibility() == Visibility.PUBLIC){
+                eventsFilteredVisibility.add(e);
+                continue;
+            }
+            if(e.getPromoter().getId() == user.getId()) {
+                eventsFilteredVisibility.add(e);
+                continue;
+            }
```

Comment:
Tu as pas moyen de faire ça 
```suggestion
            if(e.getVisibility() == Visibility.PUBLIC || e.getPromoter().getId() == user.getId()) {
                eventsFilteredVisibility.add(e);
            }
```
plutôt que de mettre des continue ?

Reply:
Fixed